src/lib/service/attest.ts

```diff
@@ -0,0 +1,99 @@
+import { encodeFunctionData } from "viem";
+
+import { TRUSTFUL_SMART_CONTRACT_ADDRESS } from "../client/constants";
+import type { Attestation } from "../shared/types";
+import { publicClient } from "../wallet/wallet-config";
+
+export interface ConnetedWalletConfiguration {
+  walletClient: any;
+  chain: number;
+}
+
+export async function submitAttest(
+  attest: Attestation,
+  configurations: ConnetedWalletConfiguration,
+  msgValue: bigint,
+) {
+  const data = encodeFunctionData({
+    abi: [
+      {
+        type: "function",
+        name: "attest",
+        inputs: [
+          {
+            name: "attestation",
+            type: "tuple",
+            internalType: "struct Attestation",
+            components: [
+              { name: "uid", type: "bytes32", internalType: "bytes32" },
+              { name: "schema", type: "bytes32", internalType: "bytes32" },
+              { name: "time", type: "uint64", internalType: "uint64" },
+              {
+                name: "expirationTime",
+                type: "uint64",
+                internalType: "uint64",
+              },
+              {
+                name: "revocationTime",
+                type: "uint64",
+                internalType: "uint64",
+              },
+              { name: "refUID", type: "bytes32", internalType: "bytes32" },
+              { name: "recipient", type: "address", internalType: "address" },
+              { name: "attester", type: "address", internalType: "address" },
+              { name: "revocable", type: "bool", internalType: "bool" },
+              { name: "data", type: "bytes", internalType: "bytes" },
+            ],
+          },
+        ],
+        outputs: [{ name: "", type: "bool", internalType: "bool" }],
+        stateMutability: "payable",
+      },
+    ],
+    args: [
+      uid,
```

Comment:
These values are missing the `attest` on front, and the name of the variable itself should be changed to `attestation` for better legibility.

The args will be:
`attestation.uid`
`attestation.schema` 
... so on..